libs/@local/graph/store/src/filter/mod.rs

```diff
@@ -606,6 +609,141 @@ impl TryFrom<PolicyExpressionTree> for Filter<'_, EntityTypeWithMetadata> {
     }
 }
 
+#[derive(Debug, derive_more::Display)]
+#[display("expression is not supported: {_0:?}")]
+pub struct InvalidPermissionCondition(PermissionCondition);
+
+impl Error for InvalidPermissionCondition {}
+
+impl TryFrom<PermissionCondition> for Filter<'_, Entity> {
+    type Error = Report<InvalidPermissionCondition>;
+
+    fn try_from(expression: PermissionCondition) -> Result<Self, Self::Error> {
+        match expression {
+            PermissionCondition::All(expressions) => Ok(Self::All(
+                expressions
+                    .into_iter()
+                    .map(Self::try_from)
+                    .collect::<Result<_, _>>()?,
```

Comment:
non-blocking - but out of curiosity - if you use `try_collect` instead of `collect` - would you still need to use turbofish syntax or could we completely forgo adding the type?

Reply:
The fish can be omitted with `try_collect`, but I did not use `try_collect` at all, yet, as it's unstable. (Also don't feel against using it, but we should do it in more locations then by searching for `collect::<Result` I guess.